src/Dhall/TH.hs

```diff
@@ -0,0 +1,28 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module Dhall.TH where
+
+import           Control.Monad
+import qualified Data.Text.Lazy as Text
+import           Data.Typeable
+import qualified Dhall.Core as D
+import qualified Dhall.Import as D
+import qualified Dhall.Parser as D
+import qualified Dhall.TypeCheck as D
+import           Language.Haskell.TH.Syntax
+
+-- A workaround for a problem in TemplateHaskell (see
+-- https://stackoverflow.com/questions/38143464/cant-find-inerface-file-declaration-for-variable)
+liftText :: Text.Text -> Q Exp
+liftText = fmap (AppE (VarE 'Text.pack)) . lift . Text.unpack
+
+importDhallExpression :: Text.Text -> IO (D.Expr D.Src D.X)
+importDhallExpression =
+  either (const (fail "invalid Dhall expression")) D.load
```

Comment:
Perhaps include the name `Dhall.TH.importDhallExpression` in the error string so that it's easier for people to hunt down the source for where the error originates from